instruction-error/src/lib.rs

```diff
@@ -202,15 +173,7 @@ pub enum InstructionError {
     IncorrectAuthority,
 
     /// Failed to serialize or deserialize account data
-    ///
-    /// Warning: This error should never be emitted by the runtime.
-    ///
-    /// This error includes strings from the underlying 3rd party Borsh crate
-    /// which can be dangerous because the error strings could change across
-    /// Borsh versions. Only programs can use this error because they are
-    /// consistent across Solana software versions.
-    ///
-    BorshIoError(String),
+    BorshIoError,
```

Comment:
Confirming that this change is good from the runtime side too -- the error text is never used or stored anywhere